pootle/static/js/shared/components/Modal.js

```diff
@@ -58,8 +82,14 @@ const Modal = React.createClass({
   /* Handlers */
 
   handleClose() {
+    if ('onCanClose' in this.props && !this.props.onCanClose()) {
```

Comment:
This provides a hook on whether the dialog can be closed. It reads: if we have `onCanClose` callback defined and it evaluates to false, then prohibit closing. This can be used to implement e.g. 'You have unsaved stuff' kind of interaction. Pretty standard technique in UI components, I'd say.